src/Microsoft.DotNet.XHarness.WebAssembly/xunit-runner.cs

```diff
@@ -0,0 +1,395 @@
+                        if (args[1] == "xunit.core")
+                            discovererType = typeof(IXunitTestCollectionFactory).Assembly.GetType(args[0]);
+                        if (discovererType == null)
+                        {
+                            Console.WriteLine("FAIL (discoverer): " + args[0] + " " + args[1]);
+                            failed = true;
+                        }
+
+                        IDataDiscoverer discoverer;
+                        discoverer = ExtensibilityPointFactory.GetDataDiscoverer(this, discovererType);
+
+                        try
+                        {
+                            IEnumerable<object[]> data = discoverer.GetData(dataAttribute, tc.TestMethod.Method);
+                            object[][] data_arr = data.ToArray();
+                            Console.WriteLine(tc.DisplayName + " [" + data_arr.Length + "]");
+                            foreach (object[] dataRow in data_arr)
+                            {
+                                nrun++;
+                                object obj = null;
+                                if (!method.IsStatic)
+                                {
+                                    ConstructorInfo constructor = method.ReflectedType.GetConstructor(Type.EmptyTypes);
+                                    if (constructor != null)
+                                        obj = constructor.Invoke(null);
+                                    else
+                                        obj = System.Runtime.Serialization.FormatterServices.GetUninitializedObject(method.ReflectedType);
+                                }
+
+                                ParameterInfo[] pars = method.GetParameters();
+                                for (int i = 0; i < dataRow.Length; ++i)
+                                    dataRow[i] = ConvertArg(dataRow[i], pars[i].ParameterType);
+                                method.Invoke(obj, BindingFlags.Default, null, dataRow, null);
+                            }
+                        }
+                        catch (Exception ex)
+                        {
+                            Console.WriteLine("FAIL: " + tc.DisplayName + " " + ex);
+                            failed = true;
+                        }
+                    }
+                    if (failed)
+                    {
+                        nfail++;
+                        continue;
+                    }
+                }
+                else
+                {
+                    if (!string.IsNullOrEmpty(tc.SkipReason))
+                    {
+                        nskipped++;
+                        continue;
+                    }
+                    nrun++;
+                    string name = tc.DisplayName;
+                    if (name.Contains("("))
+                        name = name.Substring(0, name.IndexOf("("));
+                    if (name != last_name)
```

Comment:
last_name is just set to string.Empty once and used here, we could just do a != string.Empty.

Reply:
It's set outside the loop and can change on each iteration